.travis/build.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+set -ex # exit with nonzero exit code if anything fails
+
+env
+unset SPIFFY
+unset ESPTOOL2
+
+export SMING_HOME=$TRAVIS_BUILD_DIR/Sming
+
+# Check coding style
+if [ "$TRAVIS_BUILD_STAGE_NAME" == "Test" ]; then
+ 	.travis/tools/clang/format-pr.sh;
+fi
+
+# Setup ARCH SDK paths
+if [ "$SMING_ARCH" == "Esp8266" ]; then
+	export ESP_HOME=$TRAVIS_BUILD_DIR/opt/esp-alt-sdk
+	if [ "$SDK_VERSION" == "3.0.0" ]; then
+		export SDK_BASE=$SMING_HOME/third-party/ESP8266_NONOS_SDK
+	fi
+
+	export PATH=$PATH:$ESP_HOME/xtensa-lx106-elf/bin:$ESP_HOME/utils/
+fi
+
+# Full compile checks please
+export STRICT=1
+
+# Diagnostic info
+cd $SMING_HOME
+make help
+make list-config
+
+# Build the framework
+make
```

Comment:
We could try also parallel build splitting this command to the following two:
```
make submodules
make -j
```

Reply:
We could.... I gave it a brief try with appveyor but after one initial failure didn't bother any further as I figured what we really need is a clear and accurate log for when it goes wrong. I had a hell of a job with the initial CI updates on appveyor, had to remote terminal into the server to find out how to fix things so really rather not have to do that if poss.!

The `--output-sync` option might help but haven't tried it - I have make 3.81 under MinGW which doesn't have the option anyway.

I suspect on a VM parallel builds wouldn't help much anyway. My Linux install is running under VirtualBox and really doesn't like parallel builds at all, but that's my problem!

I suspect we'd also need lwip:

```
make submodules
make lwip
make -j
```